### PR TITLE
chore: [BREAKING CHANGE] upgrade tekton/pipeline to v0.56

### PR DESCRIPTION
**Pull request by l-qing**
# Changes
* chore: upgrade tekton/pipeline to v0.56
* feat: add feature flags for tekton

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [`spec` PR link](https://github.com/katanomi/spec) included
- [ ] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [ ] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [x] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

### CODE REVIEW

File: apis/applications/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ limitations under the License.
 package v1alpha1
 
 import (
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (l-qing) [inline apis/applications/v1alpha1/zz_generated.deepcopy.go:25]: make 自动修改的

File: apis/meta/v1alpha1/condition_test.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ import (
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 )
 
-//go:generate ../../../bin/mockgen -package=apis -destination=../../../testing/mock/knative.dev/pkg/apis/condition_manager.go  knative.dev/pkg/apis ConditionManager
+//go:generate mockgen -package=apis -destination=../../../testing/mock/knative.dev/pkg/apis/condition_manager.go  knative.dev/pkg/apis ConditionManager
```
Comment (l-qing) [inline apis/meta/v1alpha1/condition_test.go:37]: 之前的方式无法生成了

File: go.mod [hunk 3]
```diff
@@ -184,10 +187,18 @@ require (
...
+	k8s.io/utils v0.0.0-20230505201702-9f6742963106
 )
 
 replace (
 	go.uber.org/zap => github.com/katanomi/zap v1.18.2 // indirect
-	knative.dev/pkg => github.com/katanomi/knative-pkg v0.0.0-20230205123800-ec0dda2ef42f
+	knative.dev/pkg => github.com/katanomi/knative-pkg v0.0.0-20240111123040-cfe2d21e1b20
+)
+
+//  TODO: Remove this once https://github.com/knative/pkg/issues/2759 is fixed
```
Comment (l-qing) [inline go.mod:198]: 这个是从 tekton 的代码中拷贝的。
https://github.com/tektoncd/pipeline/blob/378c61e7449425d61878a12e94871efdd2729697/go.mod#L41-L47

File: multicluster/filter_test.go [hunk 4]
```diff
@@ -52,7 +53,9 @@ var _ = Describe("TestCrossClusterSubjectReview_GetClient", func() {
 			ClusterNamespace: "test",
 			restMapper:       meta.NewDefaultRESTMapper([]schema.GroupVersion{}),
 		}
-		req = restful.NewRequest(&http.Request{})
+		req = restful.NewRequest(&http.Request{
```
Comment (l-qing) [inline multicluster/filter_test.go:56]: 依赖更新后，原来会 panic 了。

File: tekton/featureflags.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+// WithEnableCELInWhenExpression enables the CEL in when expression
+func WithEnableCELInWhenExpression(enabled bool) ConfigOptions {
+	return func(cfg *config.Config) {
+		cfg.FeatureFlags.EnableCELInWhenExpression = enabled
+	}
+}
+
+// WithDefaultConfig sets the default config
+func WithDefaultConfig(ctx context.Context, options ...ConfigOptions) context.Context {
```
Comment (l-qing) [inline tekton/featureflags.go:78]: 快速为我们自己设置一些 config。

File: tekton/param_conversion.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+package tekton
+
+import (
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+// ConvertParamBeta1ToV1 convert param from v1beta1 to v1
+func ConvertParamBeta1ToV1(param pipelinev1beta1.Param) pipelinev1.Param {
```
Comment (l-qing) [inline tekton/param_conversion.go:25]: resolutionrequest 资源改了默认的参数版本，所以我们使用 resolutionrequest 的得转换一下。

File: tekton/param_conversion_test.go [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+)
+
+var _ = Describe("Test.Params.Conversion", func() {
+	var (
+		paramsBeta1 = []pipelinev1beta1.Param{}
+		paramsV1    = []pipelinev1.Param{}
+	)
+
+	BeforeEach(func() {
+		// loadV1Params("testdata/params.v1beta1.yaml")
```
Comment (yuzp1996) [inline tekton/param_conversion_test.go:38]: 没用就去掉

**Review by yuzp1996 — DISMISSED** [review]

**Review by yuzp1996 — APPROVED** [review]

**Pull request merged**